package/MDAnalysis/core/Selection.py

```diff
@@ -624,90 +602,115 @@ class BaseSelection(NucleicSelection):
-            if type(res) == tuple:
-                res = res[0]
-            result_set = [group.atoms[i] for i in res]
-        elif self.prop == "mass":
-            result_set = [a for a in group.atoms
-                          if self.operator(a.mass, self.value)]
-        elif self.prop == "charge":
-            result_set = [a for a in group.atoms
-                          if self.operator(a.charge, self.value)]
-        return set(result_set)
-
-
-class SameSelection(Selection):
+        self.abs_ = abs_
+
+    def apply(self, group):
+        try:
+            col = {'x':0, 'y':1, 'z':2}[self.prop]
+        except KeyError:
+            pass
+        else:
+            values = group.positions[:,col]
+
+        if self.prop == 'mass':
+            values = group.masses
+        elif self.prop == 'charge':
+            values = group.charges
+
+        if self.abs_:
+            values = np.abs(values)
+        mask = self.operator(values, self.value)
+
+        return unique(group[mask])
+
+
+class SameSelection(object):
     # When adding new keywords here don't forget to also add them to the
     #  case statement under the SAME op, where they are first checked.
     def __init__(self, sel, prop):
         self.sel = sel
         self.prop = prop
 
-    def _apply(self, group):
-        res = self.sel._apply(group)
+    def apply(self, group):
+        prop_trans = {
+            'residue':'resids',
+            'segment':'segids',
+            'name': 'names',
+            'type': 'types',
+            'resname': 'resnames',
+            'resid': 'resids',
+            'segid': 'segids',
+            'mass': 'masses',
+            'charge': 'charges',
+            'radius': 'radii',
+            'bfactor': 'bfactors',
+            'resnum': 'resnums',
+        }
+
+        res = self.sel.apply(group)
         if not res:
-            return set([])
-        if self.prop in ("residue", "fragment", "segment"):
-            atoms = set([])
-            for a in res:
-                if a not in atoms:
-                # This shortcut assumes consistency that all atoms in a residue/fragment/segment
-                # belong to the exact same residue/fragment/segment.
-                    atoms |= set(getattr(a, self.prop).atoms)
-            return Selection._group_atoms & atoms
-        elif self.prop in ("name", "type", "resname", "resid", "segid", "mass", "charge", "radius", "bfactor", "resnum"):
-            props = [getattr(a, self.prop) for a in res]
-            result_set = (a for a in Selection._group_atoms if getattr(a, self.prop) in props)
-        elif self.prop in ("x", "y", "z"):
-            p = getattr(Selection.coord, "_"+self.prop)
-            res_indices = np.array([a.index for a in res])
-            sel_indices = np.array([a.index for a in Selection._group_atoms])
-            result_set = group.atoms[np.where(np.in1d(p[sel_indices], p[res_indices]))[0]]._atoms
+            return group[[]]  # empty selection
```

Comment:
It's a list but yeah.  It does a list comprehension over the (empty) list, and returns that.  Seems weird but it copies np behaviour
```
In [1]: import numpy as np

In [2]: a = np.arange(10)

In [3]: a[[]]
Out[3]: array([], dtype=int64)
```